samples/excel/30-range/conditional-formatting-basic.yaml

```diff
@@ -0,0 +1,390 @@
+
+                await context.sync();
+            });
+        }
+
+        async function applyTextFormat() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                const range = sheet.getRange("B16:D18");
+                const conditionalFormat = range.conditionalFormats
+                    .add(Excel.ConditionalFormatType.containsText);
+                conditionalFormat.textComparison.format.font.color = "red";
+                conditionalFormat.textComparison.rule = {operator: Excel.ConditionalTextOperator.contains, text: "Delayed"};
+
+                await context.sync();
+            });
+        }
+
+        async function applyCellValueFormat() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                const range = sheet.getRange("B21:E23");
+                const conditionalFormat = range.conditionalFormats
+                    .add(Excel.ConditionalFormatType.cellValue);
+                conditionalFormat.cellValue.format.font.color = "red";
+                conditionalFormat.cellValue.rule = { formula1: "=0", operator: "LessThan" };
+
+                await context.sync();
+            });
+        }
+
+        async function applyCustomFormat() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                const range = sheet.getRange("B8:E13");
+                const conditionalFormat = range.conditionalFormats.add(Excel.ConditionalFormatType.custom);
+                conditionalFormat.custom.rule.formula = '=IF(B8>INDIRECT("RC[-1]",0),TRUE)';
+                conditionalFormat.custom.format.font.color = "green";
+
+                await context.sync();
+            });
+        }
+
+        async function listConditionalFormats() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                sheet.load("name");
+                const worksheetRange = sheet.getRange();
+                worksheetRange.load("conditionalFormats/type");
+
+                await context.sync();
+
+                $("#conditional-formats li").remove();
+                if (worksheetRange.conditionalFormats.items.length > 0) {
+                    for (let i = 0; i < worksheetRange.conditionalFormats.items.length; i++) {
+                        let $a = $(`<a href=''>${worksheetRange.conditionalFormats.items[i].type}</a>`)
+                            .click(
+                                async (e: any) => {
+                                    e.preventDefault();
+                                    const sheet = context.workbook.worksheets
+                                        .getItem(e.data.worksheet);
+                                    const conditionalFormat = sheet.getRange()
```

Comment:
Not in love with getItemAt if we can avoid it.  Can we?

Reply:
I don't see an obvious way to not use it. In fact, this looks like a scenario that getItemAt was invented for. ConditionalFormats don't have a name property or any other intrinsically unique property. BTW, what's wrong with getItemAt?